### ISSUE TITLE
Azure.APIM.HTTPBackend reports failed when service URL is not defined

### ISSUE THREAD

**Issue by BernieWhite** [Azure/PSRule.Rules.Azure#1555]
**Description of the issue**

When `properties.serviceUrl` of an APIM API sub-resource `Microsoft.ApiManagement/service/apis` is not defined `Azure.APIM.HTTPBackend` fails.

The recommendation `Use HTTPS for communication to backend services.` is reported.

**Expected behaviour**

When the `serviceUrl` is not set a HTTP service is not used so this rule should not fail.

**Module in use and version:**

- Module: PSRule.Rules.Azure
- Version: **1.18.0**

Captured output from `$PSVersionTable`:

```text
Name                           Value
----                           -----
PSVersion                      7.2.5
PSEdition                      Core
GitCommitId                    7.2.5
OS                             Microsoft Windows 10.0.22000
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```
